### PR TITLE
Grafana/ui: Add UserIcon and UsersIndicator components

### PR DESCRIPTION
**Pull request by Clarity-89**
**What is this feature?**

Move the `UserIcon` and `UsersIndicator` components into grafana/ui. 

**Why do we need this feature?**

Enable this component to be used by other parts of Grafana + plugins.

**Which issue(s) does this PR fix?**:

Part of https://github.com/grafana/grafana/issues/66679.

**Special notes for your reviewer:**

Default state:
Screenshot 2023-04-19 at 13 06 02
Screenshot 2023-04-19 at 14 21 38

Active state:
Screenshot 2023-04-20 at 8 49 02

With many users
Screenshot 2023-04-20 at 8 50 00

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/UsersIndicator/UserIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+ * @param {string} [name] The name to extract initials from.
+ * @returns {string} The uppercase initials of the first and last name.
+ * @example
+ * // Returns 'JD'
+ * getUserInitials('John Doe');
+ * // Returns 'A'
+ * getUserInitials('Alice');
+ * // Returns ''
+ * getUserInitials();
+ */
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/UsersIndicator/UserIcon.tsx:44]: I love this!

File: packages/grafana-ui/src/components/UsersIndicator/types.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+export interface UserView {
+  user: {
+    /** User's name, containing first + last name */
+    name: string;
+    /** URL to the user's avatar */
+    avatarUrl?: string;
+  };
+  /** Datetime string when the user was last active */
+  lastActiveAt: DateTimeInput;
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/UsersIndicator/types.ts:11]: Does this have to be ISO UTC?

Comment (Clarity-89) [reply packages/grafana-ui/src/components/UsersIndicator/types.ts:11]: Can be anything that our `dateTime` function can convert.

File: packages/grafana-ui/src/components/UsersIndicator/UserIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  return {
+    container: css`
+      padding: 0;
+      width: 30px;
+      height: 30px;
+      background: none;
+      border: none;
+      border-radius: 50%;
+      & > * {
+        border-radius: 50%;
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/UsersIndicator/UserIcon.tsx:121]: There is a border radius token for this now, `theme.shape.radius.circle`

File: packages/grafana-ui/src/components/UsersIndicator/UserIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    `,
+    dotContainer: css`
+      display: flex;
+      align-items: center;
+    `,
+    dot: css`
+      height: 6px;
+      width: 6px;
+      background-color: ${theme.colors.primary.main};
+      border-radius: 50%;
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/UsersIndicator/UserIcon.tsx:163]: Same here about border radius token

**Review by JoaoSilvaGrafana — COMMENTED** [review]
Nice job on moving these 👍 
just a couple of comments

File: packages/grafana-ui/src/components/UsersIndicator/UsersIndicator.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  onClick?: () => void;
+}
+export const UsersIndicator = ({ users, onClick, limit = 4 }: UsersIndicatorProps) => {
+  const styles = useStyles2(getStyles);
+  if (!users.length) {
+    return null;
+  }
+
+  const limitReached = users.length > limit;
+  const extraUsers = users.length - limit;
```
[the remark this answers is not in the document]
Comment (Clarity-89) [reply packages/grafana-ui/src/components/UsersIndicator/UsersIndicator.tsx:27]: Done! Also added `min` param to the argType but it looks like it doesn't apply to manually typed values.

**Review by JoaoSilvaGrafana — APPROVED** [review]
Nice, lgtm! 👏

**Pull request merged**